### DISCUSSION TITLE
Pyo3 build script rebuilds all the time...

### DISCUSSION THREAD

**Discussion by Igosuki** [PyO3/pyo3#2088] [Questions] [answered]
Hi guys, I'm using Pyo3 to bring a rust API to Python, and when I enable the python feature in the code, pyo3 modules recompile all the time, why does it need to rebuild if the code hasn't changed ? Even if I add a single blank line or a print statement to another module the pyo3 modules still rebuild...

```
Compiling pyo3-build-config v0.15.1 (https://github.com/PyO3/pyo3?rev=8a989eeb6aeb862c0c518d994ef6e5f967f0dc46#8a989eeb)
   Compiling pyo3-macros-backend v0.15.1 (https://github.com/PyO3/pyo3?rev=8a989eeb6aeb862c0c518d994ef6e5f967f0dc46#8a989eeb)
   Compiling pyo3 v0.15.1 (https://github.com/PyO3/pyo3?rev=8a989eeb6aeb862c0c518d994ef6e5f967f0dc46#8a989eeb)
   Compiling pyo3-macros v0.15.1 (https://github.com/PyO3/pyo3?rev=8a989eeb6aeb862c0c518d994ef6e5f967f0dc46#8a989eeb)

```

**Comment by davidhewitt** [comment, marked as answer]
Does the discussion in #1708 help? In particular changing the rust-analyzer target directory as a workaround.

**Comment by Igosuki** [comment]
I did it through a config.toml file using `target-dir` but it works, thanks.
